### PR TITLE
Centrality proper migration to consumes

### PR DESCRIPTION
**Pull request by yetkinyilmaz**
CentralityProvider was incompatible with framework, removed completely.
CentralityBinProducer contains the relevant calculations inside the module.
This PR also includes the follow-up on detailed code optimization requests from @StoyanStoynev in 
https://github.com/cms-sw/cmssw/pull/5564

### CODE REVIEW

**Comment by yetkinyilmaz** [comment]
@azsigmon @BetterWang @smdogra @tuos should follow this thread closely.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoHI/HiCentralityAlgos/plugins/CentralityBinProducer.cc [hunk 1]
```diff
@@ -39,19 +44,29 @@
...
 class CentralityBinProducer : public edm::EDProducer {
+  enum VariableType {HFtowers = 0, HFtowersPlus = 1, HFtowersMinus = 2, HFtowersTrunc = 3, HFtowersPlusTrunc = 4, HFtowersMinusTrunc = 5, HFhits = 6, PixelHits = 7, PixelTracks = 8, Tracks = 9, EB = 10, EE = 11, Missing = 12};
    public:
       explicit CentralityBinProducer(const edm::ParameterSet&);
-      explicit CentralityBinProducer(const edm::ParameterSet&, const edm::EventSetup&, edm::ConsumesCollector &&);
       ~CentralityBinProducer();
 
    private:
-      virtual void beginJob() override ;
+      virtual void beginRun(edm::Run const& run, const edm::EventSetup& iSetup);
```
Comment (slava77) [inline RecoHI/HiCentralityAlgos/plugins/CentralityBinProducer.cc]: need override in the end

**Comment by slava77** [comment]
... I've started checking this

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/7346	ffcf1eb
checked with a 400 events in 140.53: no differences in DQM plots; no clear changes in techincal performance (CPU/MEM) either.

**Comment by deguio** [comment]
+1

**Pull request merged**